application/domain/domain.go

```diff
@@ -524,3 +525,39 @@ func EmailFromAddress(name *string) string {
 	}
 	return addr
 }
+
+// RemoveUnwantedChars removes characters from `str` that are not in `allowed` and not in "safe" character ranges.
+func RemoveUnwantedChars(str, allowed string) string {
+	return strings.Map(func(r rune) rune {
```

Comment:
I think this would be more readable to me ...
```
	removeFunc := func(r rune) rune {
		if strings.IndexRune(allowed, r) >= 0 || isSafeRune(r) {
			return r
		}
		return -1
	}
	return strings.Map(removeFunc, str)
```

Reply:
That is better.